syft/format/syftjson/to_format_model.go

```diff
@@ -278,6 +293,20 @@ func toPackageModel(p pkg.Package, cfg EncoderConfig) model.Package {
 	}
 }
 
+func toLocationsModel(locations file.LocationSet, layerOrderByDigest map[string]int) []file.Location {
```

Comment:
What if instead of dealing directly with the `layerOrderByDigest` this took a type-safe sort func, e.g.: `func(a, b file.Location) int`. This could allow us to construct a layer sorter and just pass the sort func reference, e.g. something along these lines:
```
func LayerSorter(layers []source.LayerMetadata) func(a, b file.Location) int
```
... and users call `slices.Sort` with it